components/common-go/baseserver/testing.go

```diff
@@ -35,6 +35,33 @@ func NewForTests(t *testing.T, opts ...Option) *Server {
 	return srv
 }
 
+func MustUseRandomLocalAddress(t *testing.T) *ServerConfiguration {
+	t.Helper()
+
+	return &ServerConfiguration{
+		Address: fmt.Sprintf("localhost:%d", MustFindFreePort(t)),
+	}
+}
+
+func MustFindFreePort(t *testing.T) int {
```

Comment:
I think this is possibly racy. We're binding a socket to find out it's free, but then we release it immediately and again re-bind during server startup. During tests, we can have a number of servers running concurrently and do not have guarantees in which order they will execute.

I think here we should either make it so we always require an explicit port and never assign one automatically, it then becomes the Test's problem to find an available port, or support binding the socket only once during server startup.

Reply:
This is indeed racy. Short of exposing the listener through API I fret there's no good way to not make it racy. 

> I think here we should either make it so we always require an explicit port and never assign one automatically

That's what's implemented now. In addition there's the helper func to try and find a free port. I'd vote for going with this for - if the raciness proves to be a problem in practice, we extend the API so that tests can hand in a bound listener.

Reply:
Let's see how this goes. I think testing with a baseserver is already an improvement compared to what we have today for most things. We can always follow-up on this.